File: SkyMood.Core/WeatherCodes.cs
namespace SkyMood;

/// <summary>
/// Maps WMO weather-interpretation codes (the scheme Open-Meteo and most providers use) to a
/// friendly label, an emoji, a scene category, and rain/snow intensities for the animation.
/// </summary>
public static class WeatherCodes
{
    public static WeatherVisual Describe(int code) => code switch
    {
        0       => new("Clear sky",          "☀️", WeatherKind.Clear,     0,   0),
        1       => new("Mainly clear",        "🌤️", WeatherKind.FewClouds, 0,   0),
        2       => new("Partly cloudy",       "⛅",  WeatherKind.FewClouds, 0,   0),
        3       => new("Overcast",            "☁️", WeatherKind.Cloudy,    0,   0),
        45      => new("Foggy",               "🌫️", WeatherKind.Fog,       0,   0),
        48      => new("Rime fog",            "🌫️", WeatherKind.Fog,       0,   0),
        51      => new("Light drizzle",       "🌦️", WeatherKind.Rain,      30,  0),
        53      => new("Drizzle",             "🌦️", WeatherKind.Rain,      45,  0),
        55      => new("Heavy drizzle",       "🌧️", WeatherKind.Rain,      60,  0),
        56      => new("Freezing drizzle",    "🌧️", WeatherKind.Rain,      45,  0),
        57      => new("Freezing drizzle",    "🌧️", WeatherKind.Rain,      55,  0),
        61      => new("Light rain",          "🌧️", WeatherKind.Rain,      50,  0),
        63      => new("Rain",                "🌧️", WeatherKind.Rain,      80,  0),
        65      => new("Heavy rain",          "🌧️", WeatherKind.Rain,      120, 0),
        66      => new("Freezing rain",       "🌧️", WeatherKind.Rain,      70,  0),
        67      => new("Freezing rain",       "🌧️", WeatherKind.Rain,      90,  0),
        71      => new("Light snow",          "🌨️", WeatherKind.Snow,      0,   30),
        73      => new("Snow",                "❄️", WeatherKind.Snow,      0,   55),
        75      => new("Heavy snow",          "❄️", WeatherKind.Snow,      0,   90),
        77      => new("Snow grains",         "🌨️", WeatherKind.Snow,      0,   40),
        80      => new("Light showers",       "🌦️", WeatherKind.Rain,      50,  0),
        81      => new("Showers",             "🌧️", WeatherKind.Rain,      80,  0),
        82      => new("Violent showers",     "⛈️", WeatherKind.Storm,     130, 0),
        85      => new("Snow showers",        "🌨️", WeatherKind.Snow,      0,   60),
        86      => new("Heavy snow showers",  "❄️", WeatherKind.Snow,      0,   100),
        95      => new("Thunderstorm",        "⛈️", WeatherKind.Storm,     90,  0),
        96      => new("Thunderstorm + hail", "⛈️", WeatherKind.Storm,     110, 0),
        99      => new("Severe thunderstorm", "⛈️", WeatherKind.Storm,     140, 0),
        _       => new("Unknown",             "🌡️", WeatherKind.Cloudy,    0,   0),
    };
}

File: SkyMood.Watch/SkyMoodWatch/WeatherCodes.swift
import Foundation

/// Mirrors SkyMood.Core/WeatherCodes.cs so the watch shows the same label + emoji as the phone app.
enum WeatherCodes {
    static func describe(_ code: Int) -> (label: String, emoji: String) {
        switch code {
        case 0: return ("Clear sky", "☀️")
        case 1: return ("Mainly clear", "🌤️")
        case 2: return ("Partly cloudy", "⛅")
        case 3: return ("Overcast", "☁️")
        case 45: return ("Foggy", "🌫️")
        case 48: return ("Rime fog", "🌫️")
        case 51: return ("Light drizzle", "🌦️")
        case 53: return ("Drizzle", "🌦️")
        case 55: return ("Heavy drizzle", "🌧️")
        case 56, 57: return ("Freezing drizzle", "🌧️")
        case 61: return ("Light rain", "🌧️")
        case 63: return ("Rain", "🌧️")
        case 65: return ("Heavy rain", "🌧️")
        case 66, 67: return ("Freezing rain", "🌧️")
        case 71: return ("Light snow", "🌨️")
        case 73: return ("Snow", "❄️")
        case 75: return ("Heavy snow", "❄️")
        case 77: return ("Snow grains", "🌨️")
        case 80: return ("Light showers", "🌦️")
        case 81: return ("Showers", "🌧️")
        case 82: return ("Violent showers", "⛈️")
        case 85: return ("Snow showers", "🌨️")
        case 86: return ("Heavy snow showers", "❄️")
        case 95: return ("Thunderstorm", "⛈️")
        case 96: return ("Thunderstorm + hail", "⛈️")
        case 99: return ("Severe thunderstorm", "⛈️")
        default: return ("Unknown", "🌡️")
        }
    }
}
